### PR TITLE
fix(poolops): reject KES issue counter overflow

### PR DESCRIPTION
**Pull request by wolf31o2**
## Summary

- Reject KES issue-number overflow before rotating an operational certificate.
- Return the existing invalid-request error and cover the boundary with a regression test.

## Validation

- Focused poolops tests and race tests passed.
- Root `go test -race ./...` passed.
- `git diff --check` passed.

This is a bounded DB-ACT-13 fix. The report’s complete key-type, official-vector, downstream-consumer, and release-chain validation remains open.

---
## Summary by cubic
Rejects KES issue counter overflow, malformed VRF/KES export material, invalid pool certificate inputs, and unsafe key files so counters can't wrap to zero, malformed keys fail early, bad margins return errors instead of panicking, and secret keys aren't loaded from non-regular or permissive files.

- Returns `ErrInvalidRequest` when the previous issue number is `math.MaxUint64`.
- Validates VRF and KES key sizes and KES depth in export helpers.
- Validates pool registration and retirement certificate inputs: nil pointers and margin bounds.
- Hardens key file loading: rejects symlinks and reparse points, caps file size, and enforces restrictive permissions on secret keys.
- Adds regression tests for overflow, invalid VRF/KES material, invalid margins, and file loading edge cases, plus an authoritative envelope vector suite covering all exported key types with trailing-CBOR, wrong-length, and mismatched-identity rejection.

<sup>Written for commit 19bdf28f1ae513f499fe9d10b6da1f435d11903e. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/blinklabs-io/bursa/pull/839?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: ui/internal/poolops/service.go [hunk 1]
```diff
@@ -288,6 +289,9 @@ func (s *Service) IssueOpCert(password string, kesIndex uint32, issueNumber, kes
 // counter incremented. The current issue number is supplied by the caller (read
 // from the previous opcert / counter file); the new cert uses prevIssue+1.
 func (s *Service) RotateKES(password string, newKESIndex uint32, prevIssueNumber, kesPeriod uint64) (OpCert, error) {
+	if prevIssueNumber == math.MaxUint64 {
```
Comment (arepala-uml) [inline ui/internal/poolops/service.go:292]: Verified: the overflow check runs before the `+1` increment, so `prevIssueNumber` can't already have wrapped when this fires. `math.MaxUint64` matches the field's real width all the way through `IssueOpCert` -> `CreateOperationalCertificate` -> `ledger.CreateOpCert` (uint64 throughout, no narrowing cast), so this is the true boundary, not an arbitrarily-large stand-in. The regression test drives the counter to that exact value. Root-cause fix, correctly placed.

File: key_file_test.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+	wallet, err := NewWallet(testSecretKeyMnemonic)
+	require.NoError(t, err)
+	data, err := json.Marshal(wallet.PaymentSKey)
+	require.NoError(t, err)
+	path := filepath.Join(t.TempDir(), "payment.skey")
+	require.NoError(t, os.WriteFile(path, data, 0o644))
+
+	key, err := LoadKeyFromFile(path)
+	assert.Nil(t, key)
+	assert.ErrorIs(t, err, ErrInsecureFileMode)
```
Comment (arepala-uml) [inline key_file_test.go]: Non-blocking, re: cubic's finding on this line. Confirmed valid by reading the platform split: `key_permissions_unix.go`/`key_permissions_windows.go` implement `checkOpenFilePermissions` very differently (POSIX mode bits vs. DACL), and this new file carries no `//go:build unix` tag, unlike the existing `key_file_permissions_unix_test.go` (which does tag `unix`). So `TestLoadKeyFromFileRejectsPermissiveSecret` asserts a Unix-only outcome (`0o644` -> `ErrInsecureFileMode`) that doesn't hold under Windows' DACL-inheritance-from-tempdir behavior.

This can't fail CI today: `go-test.yml`'s matrix is `platform: [ubuntu-latest]` only, and the `windows` entries that do exist (cross-build, publish's build-msi) only `go build`, never `go test`. So it's a latent portability/flakiness issue, not a merge blocker. Recommend moving this one assertion under the existing unix tag (or into `key_file_permissions_unix_test.go` alongside `TestReadSecretKeyFileRejectsFIFO`) before Windows `go test` coverage is ever added.

**Review by arepala-uml — DISMISSED** [review]

**Review by arepala-uml — APPROVED** [review]
